storage/sis-storage/src/main/java/org/apache/sis/storage/tilematrix/TileMatrix.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.sis.storage.tilematrix;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.apache.sis.coverage.grid.GridExtent;
+import org.apache.sis.coverage.grid.GridGeometry;
+import org.apache.sis.storage.DataStoreException;
+import org.opengis.util.GenericName;
+
+/**
+ * A TileMatrix is collection of tiles with the same size and properties placed
+ * on a regular grid with no overlaping.
+ * While a TileMatrix in a regular 2D grid space it may still be defined on other dimensions,
+ * it's GridGeometry may have any number of additional axes.
+ *
+ * @author Johann Sorel (Geomatys)
+ */
+public interface TileMatrix {
+
+    /**
+     * @return unique identifier.
+     */
+    GenericName getIdentifier();
+
+    /**
+     * Returns the structure of the tile matrix.
+     * Returned {@linkplain GridGeometry} is always a two dimensional slice, it may still contain other dimension such as time and
```

Comment:
I propose to remove the restriction to two-dimensional slices, and allow `TileMatix` to have any number of dimensions.